File: app/src/main/java/com/anenha/weather/app/entity/ForecastEntity.java
package com.anenha.weather.app.entity;

import android.content.Context;
import android.graphics.drawable.Drawable;
import com.anenha.weather.app.model.Condition;
import com.anenha.weather.app.model.Forecast;
import com.anenha.weather.app.utils.Formatter;


/**
 * Created by ajnen on 12/10/2017.
 */

public class ForecastEntity extends CoreEntity {
    private Context context;
    private Forecast forecast;

    private Drawable image;
    private String date;
    private String day;
    private String high;
    private String low;
    private String description;

    public ForecastEntity(final Context context, final Forecast forecast) {
        super(context);
        this.context = context;
        this.forecast = forecast;

        setImage();
        setDay();
        setDate();
        setHigh();
        setLow();
        setText();
    }

    private void setImage() {
        int resourceId = context.getResources().getIdentifier("drawable/icon_" + forecast.getCode(), null, context.getPackageName());
        this.image = context.getDrawable(resourceId);
    }

    private void setDay() {
        this.day = Formatter.weekDay(context, forecast.getDay());
    }

    private void setDate() {
        this.date = Formatter.date(context, null, forecast.getDate(), true, false);
    }

    private void setHigh() {
        this.high = Formatter.temperature(Integer.parseInt(forecast.getHigh()), getTemperatureUnit());
    }

    private void setLow() {
        this.low = Formatter.temperature(Integer.parseInt(forecast.getLow()), getTemperatureUnit());
    }

    private void setText() {
        Condition condition = new Condition();
        condition.setCode(forecast.getCode());
        condition.setText(forecast.getText());
        this.description = Formatter.conditionDescription(context, condition);
    }

    public Drawable getImage() {
        return image;
    }

    public String getDate() {
        return date;
    }

    public String getDay() {
        return day;
    }

    public String getHigh() {
        return high;
    }

    public String getLow() {
        return low;
    }

    public String getDescription() {
        return description;
    }
}

File: app/src/main/java/com/anenha/weather/entity/ForecastEntity.kt
package com.anenha.weather.entity

import android.content.Context
import android.graphics.drawable.Drawable
import com.anenha.weather.model.ConditionModel
import com.anenha.weather.model.ForecastModel
import com.anenha.weather.utils.Formatter


/**
 * Created by ajnen on 12/10/2017.
 */

class ForecastEntity(private val context: Context, private val forecast: ForecastModel) : CoreEntity(context) {
    var image: Drawable? = null
    var date: String = ""
    var day: String = ""
    var high: String = ""
    var low: String = ""
    var description: String = ""

    init {
        setImage()
        setDay()
        setDate()
        setHigh()
        setLow()
        setText()
    }

    private fun setImage() {
        val resourceId = context.resources.getIdentifier("drawable/weather_icon_" + forecast.code, null, context.packageName)
        this.image = context.getDrawable(resourceId)
    }

    private fun setDay() {
        this.day = Formatter.weekDay(context, forecast.day!!)
    }

    private fun setDate() {
        this.date = Formatter.date(context, "", forecast.date!!, true, false)
    }

    private fun setHigh() {
        this.high = Formatter.temperature(Integer.parseInt(forecast.high), temperatureUnit)
    }

    private fun setLow() {
        this.low = Formatter.temperature(Integer.parseInt(forecast.low), temperatureUnit)
    }

    private fun setText() {
        val condition = ConditionModel()
        condition.code = forecast.code
        condition.text = forecast.text
        this.description = Formatter.conditionDescription(context, condition)
    }
}
